src/lsm/lsm_tree.c

```diff
@@ -771,6 +771,7 @@ __wt_lsm_tree_switch(WT_SESSION_IMPL *session, WT_LSM_TREE *lsm_tree)
 	++lsm_tree->dsk_gen;
 
 	lsm_tree->modified = true;
+	WT_FULL_BARRIER();
```

Comment:
@sueloverso I'd expected the barrier to be before updating the modified state - is there a reason why it should be after?